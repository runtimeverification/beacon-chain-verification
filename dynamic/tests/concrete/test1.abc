stateTransition(#Block((1,10001),(0,    0),.Slashings,#Attestation(0,0,(0,0),(0,10001),0,0,0) .Attestations,,.VoluntaryExits))
stateTransition(#Block((3,10003),(1,10001),.Slashings,.Attestations,,.VoluntaryExits))
processSlots(4)
