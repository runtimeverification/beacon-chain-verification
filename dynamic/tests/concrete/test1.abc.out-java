<T>
  <k>
    .
  </k>
  <currentSlot>
    4
  </currentSlot>
  <states>
    <slot>
      0
    </slot> |-> <state>
      <slot>
        0
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Slashings
      </attested>
      <justified>
        0 |-> true
      </justified>
      <finalized>
        0 |-> true
      </finalized>
      <lastBlock>
        ( 0 , 0 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state>
    <slot>
      1
    </slot> |-> <state>
      <slot>
        1
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Slashings
      </attested>
      <justified>
        0 |-> true
      </justified>
      <finalized>
        0 |-> true
      </finalized>
      <lastBlock>
        ( 1 , 10001 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state>
    <slot>
      2
    </slot> |-> <state>
      <slot>
        2
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Slashings
      </attested>
      <justified>
        0 |-> true
      </justified>
      <finalized>
        0 |-> true
      </finalized>
      <lastBlock>
        ( 1 , 10001 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state>
    <slot>
      3
    </slot> |-> <state>
      <slot>
        3
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Slashings
      </attested>
      <justified>
        0 |-> true
      </justified>
      <finalized>
        0 |-> true
      </finalized>
      <lastBlock>
        ( 3 , 10003 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state>
    <slot>
      4
    </slot> |-> <state>
      <slot>
        4
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Slashings
        1 |-> .Slashings
      </attested>
      <justified>
        0 |-> true
        1 |-> false
      </justified>
      <finalized>
        0 |-> true
        1 |-> false
      </finalized>
      <lastBlock>
        ( 3 , 10003 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state>
  </states>
  <blocks>
    <bSlot>
      0
    </bSlot> |-> <block>
      <bSlot>
        0
      </bSlot>
      <bID>
        0
      </bID>
      <parent>
        ( -1 , -1 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        .Slashings
      </attestations>
      <deposits>
        .Slashings
      </deposits>
      <exits>
        .Slashings
      </exits>
    </block>
    <bSlot>
      1
    </bSlot> |-> <block>
      <bSlot>
        1
      </bSlot>
      <bID>
        10001
      </bID>
      <parent>
        ( 0 , 0 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Slashings
      </attestations>
      <deposits>
        .Slashings
      </deposits>
      <exits>
        .Slashings
      </exits>
    </block>
    <bSlot>
      3
    </bSlot> |-> <block>
      <bSlot>
        3
      </bSlot>
      <bID>
        10003
      </bID>
      <parent>
        ( 1 , 10001 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        .Slashings
      </attestations>
      <deposits>
        .Slashings
      </deposits>
      <exits>
        .Slashings
      </exits>
    </block>
  </blocks>
</T>
