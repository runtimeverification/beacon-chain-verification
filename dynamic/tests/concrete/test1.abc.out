<T>
  <k>
    .
  </k>
  <currentSlot>
    4
  </currentSlot>
  <states>
    <state>
      <slot>
        0
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
      </attested>
      <justified>
        0 |-> true
      </justified>
      <finalized>
        0 |-> true
      </finalized>
      <lastBlock>
        ( 0 , 0 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state> <state>
      <slot>
        1
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Attestations
      </attested>
      <justified>
        0 |-> true
      </justified>
      <finalized>
        0 |-> true
      </finalized>
      <lastBlock>
        ( 1 , 10001 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state> <state>
      <slot>
        2
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Attestations
      </attested>
      <justified>
        0 |-> true
      </justified>
      <finalized>
        0 |-> true
      </finalized>
      <lastBlock>
        ( 1 , 10001 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state> <state>
      <slot>
        3
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Attestations
      </attested>
      <justified>
        0 |-> true
      </justified>
      <finalized>
        0 |-> true
      </finalized>
      <lastBlock>
        ( 3 , 10003 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state> <state>
      <slot>
        4
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Attestations
        1 |-> .Attestations
      </attested>
      <justified>
        0 |-> true
        1 |-> false
      </justified>
      <finalized>
        0 |-> true
        1 |-> false
      </finalized>
      <lastBlock>
        ( 3 , 10003 )
      </lastBlock>
      <lastJustified>
        0
      </lastJustified>
      <lastFinalized>
        0
      </lastFinalized>
    </state>
  </states>
  <blocks>
    <block>
      <bSlot>
        0
      </bSlot>
      <bID>
        0
      </bID>
      <parent>
        ( -1 , -1 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        .Attestations
      </attestations>
      <deposits>
        .Deposits
      </deposits>
      <exits>
        .VoluntaryExits
      </exits>
    </block> <block>
      <bSlot>
        1
      </bSlot>
      <bID>
        10001
      </bID>
      <parent>
        ( 0 , 0 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        #Attestation ( 90000 , 0 , ( 0 , 0 ) , ( 0 , 10001 ) )  .Attestations
      </attestations>
      <deposits>
        .Deposits
      </deposits>
      <exits>
        .VoluntaryExits
      </exits>
    </block> <block>
      <bSlot>
        3
      </bSlot>
      <bID>
        10003
      </bID>
      <parent>
        ( 1 , 10001 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        .Attestations
      </attestations>
      <deposits>
        .Deposits
      </deposits>
      <exits>
        .VoluntaryExits
      </exits>
    </block>
  </blocks>
</T>
